common/auth/auth.go

```diff
@@ -2,3 +2,24 @@ package auth
 
 const StsAPIKeyHeader = "Sts-Api-Key"
 const StsTenantContextKey = "Sts-Tenant"
+
+type AuthData struct {
```

Comment:
I don't really see why this struct was introduced here. None of the functions seem to be used in the changed code.

Reply:
it's the data structure that's stored on the client Info - so it's now indeed used behind the scenes by the resource processor.  Since it's exposed by the ingestionapikey authenticator, it is a bit of an API for other otel components.